sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.scripting
+
+import org.apache.spark.sql.catalyst.analysis.UnresolvedIdentifier
+import org.apache.spark.sql.catalyst.parser.{CompoundBody, CompoundPlanStatement, SingleStatement}
+import org.apache.spark.sql.catalyst.plans.logical.{CreateVariable, DropVariable, LogicalPlan}
+import org.apache.spark.sql.catalyst.trees.Origin
+
+/**
+ * Trait for SQL Scripting interpreters.
+ */
+trait ProceduralLanguageInterpreter {
```

Comment:
we need to justify the abstraction we make. will we add different implementations of this trait?

Reply:
I don't think we will, but I don't know if there will be a use-case in the future though. We did it like this in our POC to have the same abstraction level as parsers have with `ParserInterface` in `SparkSessionsExtensions`, `BaseSessionStateBuilder`, `SessionState` etc. - we will introduce changes there to register the interpreter etc. in the PR where we introduce API changes to support scripts.
We can revert to use only `SqlScriptingInterpreter` without the interface though if you think so - and if there was ever need in the future, we can introduce this abstraction, whatever you think is better.

Reply:
let's add abstraction when needed.

Reply:
Removed the abstraction.